applications/lfric_atm/fab_lfric_atm.py

```diff
@@ -190,43 +190,85 @@ def analyse_step(self):
+                                    flags=ovewrite_debug_optimisation),
+                           AddFlags(match='$output/*conv_gr_alg_mod_psy*',
+                                    flags=ovewrite_debug_optimisation),
+                           AddFlags(match='$output/*gungho_model_mod*',
+                                    flags=ovewrite_debug_optimisation),
+                           AddFlags(match='$output/*init_aerosol_fields_alg_mod_psy*',
+                                    flags=ovewrite_debug_optimisation),
+                           AddFlags(match='$output/*jules_extra_kernel_mod*',
+                                    flags=ovewrite_debug_optimisation),]
         else:
-            no_omp = "-fno-openmp"
-            real8 = "-fdefault-real-8"
-            no_externals = []
-        path_flags = [AddFlags(
-            '$output/science/um/atmosphere/large_scale_precipitation/*',
-            [no_omp]),
-            AddFlags(match="$output/science/*", flags=[real8]),
+            no_omp = ["-fno-openmp"]
+            um_physics = ["-fdefault-real-8"]
+        path_flags += [
+            AddFlags(match='$output/science/um/atmosphere/large_scale_precipitation/*',
+            flags=no_omp),
+            AddFlags(match="$output/science/*", flags=um_physics),
             # jules and socrates are extracted in the science folder
-            AddFlags(match="$output/legacy/*", flags=[real8]),
-            AddFlags(match="$output/AC_assimilation/*", flags=[real8]),
-            AddFlags(match="$output/aerosols/*", flags=[real8]),
-            AddFlags(match="$output/atmosphere_service/*", flags=[real8]),
-            AddFlags(match="$output/boundary_layer/*", flags=[real8]),
-            AddFlags(match="$output/carbon/*", flags=[real8]),
-            AddFlags(match="$output/convection/*", flags=[real8]),
-            AddFlags(match="$output/diffusion_and_filtering/*", flags=[real8]),
-            AddFlags(match="$output/dynamics/*", flags=[real8]),
-            AddFlags(match="$output/dynamics_advection/*", flags=[real8]),
-            AddFlags(match="$output/electric/*", flags=[real8]),
-            AddFlags(match="$output/free_tracers/*", flags=[real8]),
-            AddFlags(match="$output/gravity_wave_drag/*", flags=[real8]),
-            AddFlags(match="$output/idealised/*", flags=[real8]),
-            AddFlags(match="$output/large_scale_cloud/*", flags=[real8]),
+            AddFlags(match="$output/legacy/*", flags=um_physics),
+            AddFlags(match="$output/AC_assimilation/*", flags=um_physics),
+            AddFlags(match="$output/aerosols/*", flags=um_physics),
+            AddFlags(match="$output/atmosphere_service/*", flags=um_physics),
+            AddFlags(match="$output/boundary_layer/*", flags=um_physics),
+            AddFlags(match="$output/carbon/*", flags=um_physics),
+            AddFlags(match="$output/convection/*", flags=um_physics),
```

Comment:
I just realise that it might be useful to allow several pattern in a path-specific setting. I will try to remember and implement that :)